store/sqlstore/channel_store.go

```diff
@@ -2154,6 +2158,27 @@ func (s SqlChannelStore) SearchInTeam(teamId string, term string, includeDeleted
 	})
 }
 
+func (s SqlChannelStore) SearchArchivedInTeam(teamId string, term string) (*model.ChannelList, *model.AppError) {
```

Comment:
It would be nice not to duplicate ourselves here. But I see how extending the SearchAllChannels below could be a problem. Looks like a bigger project.

Reply:
Yeah, I tried hard to be more DRY here, but had to copy some code here and there to make it work.